src/library/scala/collection/immutable/Vector.scala

```diff
@@ -45,17 +57,11 @@ object Vector extends StrictOptimizedSeqFactory[Vector] {
 
         if (knownSize == 0) empty[E]
         else if (knownSize > 0 && knownSize <= 32) {
-          val display0 = new Array[AnyRef](knownSize)
-
-          var i = 0
-          val iterator = it.iterator
-          while (iterator.hasNext) {
-            display0(i) = iterator.next().asInstanceOf[AnyRef]
-            i += 1
-          }
+          val display0 = new Array[Any](knownSize)
+          it.iterator.copyToArray(display0)
```

Comment:
why not `it.copyToArray` to benefit from fast array copying for array-based collections?